src/commands/runDraftTool/runDraftWorkflow.ts

```diff
@@ -0,0 +1,471 @@
+      return true;
+   }
+}
+
+class PromptAKSClusterSelection extends AzureWizardPromptStep<WizardContext> {
+   constructor(private az: AzApi) {
+      super();
+   }
+
+   public async prompt(wizardContext: WizardContext): Promise<void> {
+      if (wizardContext.clusterSubscription === undefined) {
+         throw Error('AKS Subscription is undefined');
+      }
+
+      if (wizardContext.clusterResourceGroup === undefined) {
+         throw Error('AKS Resource Group is undefined');
+      }
+
+      const mcs = getAysncResult(
+         this.az.listManagedClustersBySubAndRG(
+            wizardContext.clusterSubscription,
+            wizardContext.clusterResourceGroup
+         )
+      );
+      const mcToItem = (mc: ManagedClusterItem) => ({
+         label: mc.managedCluster.name || ''
+      });
+      const mcPick = await wizardContext.ui.showQuickPick(
+         sort(getAsyncOptions(mcs, mcToItem)),
+         {
+            ignoreFocusOut,
+            stepName: 'AKS Cluster',
+            placeHolder: 'AKS Cluster',
+            noPicksMessage: 'No AKS Cluster Found'
+         }
+      );
+
+      wizardContext.cluster = (await mcs).find(
+         (mc) => mcToItem(mc).label === mcPick.label
+      );
+   }
+
+   public shouldPrompt(wizardContext: WizardContext): boolean {
+      return true;
+   }
+}
+
+class PromptACRSubscriptionSelection extends AzureWizardPromptStep<WizardContext> {
+   constructor(private az: AzApi) {
+      super();
+   }
+
+   public async prompt(wizardContext: WizardContext): Promise<void> {
+      const subs = getAysncResult(this.az.listSubscriptions());
+      const subToItem = (sub: SubscriptionItem) => ({
+         label: sub.subscription.displayName || '',
+         description: sub.subscription.subscriptionId || ''
+      });
+      const subPick = await wizardContext.ui.showQuickPick(
+         sort(getAsyncOptions(subs, subToItem)),
+         {
+            ignoreFocusOut,
+            stepName: 'Azure Container Registry (ACR) Subscription',
+            placeHolder: 'ACR Subscription',
+            noPicksMessage: 'No Subscriptions found'
+         }
+      );
+
+      // if something was recently used this text is appened to the description
+      const removeRecentlyUsed = (description: string) =>
+         description.replace(' (recently used)', '');
+      wizardContext.acrSubscription = (await subs).find(
+         (sub) =>
+            subToItem(sub).description ===
+            removeRecentlyUsed(subPick.description || '')
+      );
+   }
+
+   public shouldPrompt(wizardContext: WizardContext): boolean {
+      return true;
+   }
+}
+
+class PromptACRResourceGroupSelection extends AzureWizardPromptStep<WizardContext> {
+   constructor(private az: AzApi) {
+      super();
+   }
+
+   public async prompt(wizardContext: WizardContext): Promise<void> {
+      if (wizardContext.clusterSubscription === undefined) {
+         throw Error('ACR Subscription is undefined');
```

Comment:
this error message should refer to the cluster subscription now right? since users are never actually picking the ACR subscription?

Reply:
changed